### PR TITLE
allowed calculation of symmetric and antisymmetric parts of optical c…

### PR DESCRIPTION
**Pull request by stepan-tsirkin**
…onductivity together

This is done by adding a parameter `TRtrans` which means that under action of TR the tensor is transposed (in addition to possible change of sign if  `TRodd == True`.

So, no need to write them in separate files. Also added an optical test with symmetries. Old behaviour is preserved (but not as default) and old reference files were renamed. New and old behaviour is compared - results agree. In future we may remove the old way.

### CODE REVIEW

**Comment by stepan-tsirkin** [comment]
@jaemolihm , do you have any idea why the tests failed on github? 
and the coverage report is weird, it says that `system.ws_dist_stars` is now not covered, but this PR does not seem to touch it. Any ideas?

**Comment by jaemolihm** [comment]
Summary: The problem is not related to the contents of this PR. I think it is a problem related to codecov and multiprocessing, but I have no idea what the problem is. It may be a temporary problem, but I am not sure. I will try to investigate more.

This problem is not related to this PR because I tried rerunning a job which successfully finished 20 days ago, and now its failing due to timeout.

A useful way to investigate is to run GitHub Actions in an interactive mode: https://github.com/mxschmitt/action-tmate.

<details>
  <summary>Detailed instructions</summary>
1. Add the following two lines to `.github/workflows/ci.yml`, right below `run: .github/install_script.sh`.
```
    - name: Setup tmate session
      uses: mxschmitt/action-tmate@v3
```
2. commit and push
3. Go to Actions and select a job. There will be a link for ssh in the GitHub Actions log under the item "Setup tmate session".
</details>

When I do this, I found that the job gets stuck after printing 
```
Time to read .chk : 0.032181501388549805
Time for MMN.__init__() : 0.1498262882232666 , read : 0.1470489501953125 , headstring 0.0027773380279541016
```
and before printing UHU.

Then, when I press `ctrl-C`, the following is printed and the job continues
```
^CProcess ForkPoolWorker-13:
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/multiprocessing/process.py", line 318, in _bootstrap
    util._exit_function()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/multiprocessing/util.py", line 334, in _exit_function
    _run_finalizers(0)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/multiprocessing/util.py", line 300, in _run_finalizers
    finalizer()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/multiprocessing/util.py", line 224, in __call__
    res = self._callback(*self._args, **self._kwargs)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pytest_cov/embed.py", line 99, in cleanup
    _cleanup(_active_cov)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pytest_cov/embed.py", line 85, in _cleanup
    cov.save()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 697, in save
    data = self.get_data()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 766, in get_data
    if self._collector and self._collector.flush_data():
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/collector.py", line 473, in flush_data
    self.covdata.add_lines(self.mapped_file_dict(self.data))
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 236, in _wrapped
    return method(self, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 459, in add_lines
    self._choose_lines_or_arcs(lines=True)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 518, in _choose_lines_or_arcs
    with self._connect() as con:
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 313, in _connect
    self._open_db()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 264, in _open_db
    self._read_db()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 268, in _read_db
    with self._dbs[threading.get_ident()] as db:
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 1073, in __enter__
    self._connect()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 1052, in _connect
    self.con = sqlite3.connect(self.filename, check_same_thread=False)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 605, in _on_sigterm
    self._atexit("sigterm")
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 601, in _atexit
    self.save()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 697, in save
    data = self.get_data()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 766, in get_data
    if self._collector and self._collector.flush_data():
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/collector.py", line 473, in flush_data
    self.covdata.add_lines(self.mapped_file_dict(self.data))
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/sqldata.py", line 234, in _wrapped
    with self._lock:
KeyboardInterrupt
```

So, it seems that there is some problem with multiprocessing and codecov. Without codecov, the test successes. But it prints some messages like the following, I don't know whether they are bad or not.
```
*** SIGTERM received at time=1643254963 on cpu 0 ***
PC: @     0x7f1e356ffa65  (unknown)  sem_post@@GLIBC_2.2.5
    @     0x7f1e35755210  (unknown)  (unknown)
    @     0x7f1e35c65680  (unknown)  (unknown)
[2022-01-27 03:42:43,515 E 9078 6037] logging.cc:317: *** SIGTERM received at time=1643254963 on cpu 0 ***
[2022-01-27 03:42:43,515 E 9078 6037] logging.cc:317: PC: @     0x7f1e356ffa65  (unknown)  sem_post@@GLIBC_2.2.5
[2022-01-27 03:42:43,517 E 9078 6037] logging.cc:317:     @     0x7f1e35755210  (unknown)  (unknown)
[2022-01-27 03:42:43,519 E 9078 6037] logging.cc:317:     @     0x7f1e35c65680  (unknown)  (unknown)
```

**Comment by jaemolihm** [comment]
I found the reason. It was indeed related to pytest-cov and multiprocessing. I fixed and made a PR: #132. See there for details.

File: tests/test_kubo.py [hunk 1]
```diff
@@ -52,45 +54,95 @@ def _inner(system, quantities, fout_name, Efermi, omega, grid_param, comparer,
...
         adpt_num_iter=adpt_num_iter, comparer=compare_energyresult,
-        additional_parameters=kubo_params)
+        additional_parameters=kubo_params,
+        specific_parameters = specific_parameters)
+    
+    compare_sym_asym("kubo_Fe_W90")
+
+
+def test_optical_sym(check_integrate_dynamical, system_Fe_W90, compare_energyresult,compare_sym_asym):
+    """Test optical properties: optical conductivity and spin Hall conductivity"""
```
Comment (jaemolihm) [inline tests/test_kubo.py:126]: Please mention in the comment that one is using symmetry (irreducible k).

**Review by jaemolihm — APPROVED** [review]
Looks good. Only a single minor comment.

**Pull request merged**